### PR TITLE
Add links to PennyLane Challenges page

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**

The [PennyLane Challenges](https://pennylane.ai/challenges/) page is now live!

**Description of the Change:**

This PR makes the links in the navbar and footer consistent with the PennyLane website such that:
1. There is a link to the PennyLane Challenges page in the navbar and footer.
2. There is no longer a link to the PennyLane FAQ in the footer.

It also removes the "external" icon from the API and Catalyst links in the navbar.

**Benefits:**

* The navbar and footer are consistent between the PennyLane website and the Docs pages.

**Possible Drawbacks:**

None.

**Related GitHub Issues:**

None.

---

**Preview:** https://xanaduai-pennylane--33.com.readthedocs.build/projects/sphinx-theme/en/33/.

### CODE REVIEW

**Review by doctorperceptron — APPROVED** [review]
# :partying_face:

**Review by Lorraine-at-Xanadu — APPROVED** [review]
Looks great! Thanks, @Mandrenkov.

File: pennylane_sphinx_theme/footer.py [hunk 1]
```diff
@@ -88,10 +92,6 @@
                     "name": "Glossary",
                     "href": "https://pennylane.ai/qml/glossary/",
                 },
-                {
-                    "name": "FAQ",
-                    "href": "https://pennylane.ai/faq/",
-                },
```
Comment (AndrewGardhouse) [inline pennylane_sphinx_theme/footer.py:94]: What's the reason for removing `FAQ` from the dropdown?
Screenshot 2023-09-25 at 10 34 20 AM

Comment (Mandrenkov) [reply pennylane_sphinx_theme/footer.py:94]: @AndrewGardhouse The FAQ is still in the navbar - it was just removed from the footer!

image

Comment (AndrewGardhouse) [reply pennylane_sphinx_theme/footer.py:94]: Ah gotcha 🤦 I should have read the file name... 😂

**Pull request merged**